services/cd-service/pkg/service/environment.go

```diff
@@ -17,10 +17,39 @@ Copyright 2023 freiheit.com*/
 package service
 
 import (
+	"context"
+
 	api "github.com/freiheit-com/kuberpult/pkg/api/v1"
+
 	"github.com/freiheit-com/kuberpult/services/cd-service/pkg/config"
+	"github.com/freiheit-com/kuberpult/services/cd-service/pkg/repository"
 )
 
+type EnvironmentServiceServer struct {
+	Repository repository.Repository
+}
+
+func (o *EnvironmentServiceServer) GetEnvironmentConfig(
+	ctx context.Context,
+	in *api.GetEnvironmentConfigRequest) (*api.GetEnvironmentConfigResponse, error) {
+	state := o.Repository.State()
+	config, err := state.GetEnvironmentConfig(in.Environment)
+	if config == nil {
```

Comment:
```suggestion
	if err != nil {
```

Reply:
Do we ever expect `nil, nil` or `non-nil, non-nil`? If so why prefer `err != nil`?

Reply:
done